packages/shopping-cart/src/use-shopping-cart.ts

```diff
@@ -1,11 +1,49 @@
-import { useContext } from 'react';
+import debugFactory from 'debug';
+import { useContext, useState, useEffect, useRef, useMemo } from 'react';
 import ShoppingCartContext from './shopping-cart-context';
-import type { ShoppingCartManager } from './types';
+import ShoppingCartOptionsContext from './shopping-cart-options-context';
+import useRefetchOnFocus from './use-refetch-on-focus';
+import type { UseShoppingCart } from './types';
 
-export default function useShoppingCart(): ShoppingCartManager {
-	const manager = useContext( ShoppingCartContext );
-	if ( ! manager ) {
+const debug = debugFactory( 'shopping-cart:use-shopping-cart' );
+
+export default function useShoppingCart( cartKey?: string | undefined ): UseShoppingCart {
```

Comment:
I think I had it this way because my goal is to remove the `?` in a follow-up PR once we've migrated all uses of the hook, but it can always be changed later.